minifi/minifi-commons/minifi-commons-framework/src/main/java/org/apache/nifi/minifi/commons/service/FlowEnrichService.java

```diff
@@ -90,6 +91,9 @@ public byte[] enrichFlow(byte[] flowCandidate) {
         }
 
         VersionedDataflow versionedDataflow = parseVersionedDataflow(flowCandidate);
+        versionedDataflow.setReportingTasks(ofNullable(versionedDataflow.getReportingTasks()).orElseGet(ArrayList::new));
+        versionedDataflow.setRegistries(ofNullable(versionedDataflow.getRegistries()).orElseGet(ArrayList::new));
+        versionedDataflow.setControllerServices(ofNullable(versionedDataflow.getControllerServices()).orElseGet(ArrayList::new));
```

Comment:
Do we need to do this for the others (e.g.: parameters) as well or only these are needed to be initialised on case of an empty flow

Reply:
For ParameterProviders and FlowAnalysisRules it is handled in the VersionedFlowSynchronizer